File: kotlin-completion/src/main/java/com/tyron/kotlin_completion/util/PsiUtils.java
package com.tyron.kotlin_completion.util;

import com.intellij.psi.PsiElement;
import org.jetbrains.kotlin.psi.psiUtil.PsiUtilsKt;

import kotlin.sequences.Sequence;
import kotlin.sequences.SequencesKt;

public class PsiUtils {

    public static <Find> Find findParent(PsiElement element, Class<Find> find) {
        Sequence<PsiElement> parentsWithSelf = PsiUtilsKt.getParentsWithSelf(element);
        Sequence<Find> sequence = SequencesKt.filterIsInstance(parentsWithSelf, find);
        return SequencesKt.firstOrNull(sequence);
    }
}

File: kotlin-completion/src/main/java/com/tyron/kotlin_completion/util/PsiUtils.kt
/*
 *  This file is part of CodeAssist.
 *
 *  CodeAssist is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  CodeAssist is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *   along with CodeAssist.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.tyron.kotlin_completion.util

import com.intellij.psi.PsiElement
import org.jetbrains.kotlin.psi.psiUtil.parentsWithSelf
import org.jetbrains.kotlin.psi.KtSimpleNameExpression

object PsiUtils {

    @JvmStatic
    fun findParent(element: PsiElement): KtSimpleNameExpression? {
        val parentWithSelf = element.parentsWithSelf
        val sequence = parentWithSelf.filterIsInstance(KtSimpleNameExpression::class.java)
        return sequence.firstOrNull()
    }
}